src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -0,0 +1,135 @@
+package org.allenai.plugins
+
+import sbt.{
+  relativeTo,
+  settingKey,
+  taskKey,
+  AutoPlugin,
+  Def,
+  Keys,
+  PathFinder,
+  Plugins,
+  SettingKey,
+  TaskKey
+}
+import sbt.plugins.JvmPlugin
+
+import java.io.File
+
+/** Plugin for building docker images. */
+object DockerBuildPlugin extends AutoPlugin {
+  val AI2_PRIVATE_REGISTRY = "allenai-docker-private-docker.bintray.io"
+
+  val DEFAULT_BASE_IMAGE = AI2_PRIVATE_REGISTRY + "/java:8"
+
+  /** Requires the JvmPlugin, since this will be building a jar dependency tree. */
+  override def requires: Plugins = JvmPlugin
+
+  object autoImport {
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // The following settings affect both the images and Dockerfiles generated by this plugin. When
+    // you update these settings in a build.sbt file, you'll want to re-generate your Dockerfiles.
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    val dockerfileLocation: SettingKey[File] = settingKey[File](
+      "The location of the Dockerfile to use in building the main project image. Defaults to " +
+        "`srcDirectory.value + \"docker/Dockerfile\", typically \"src/main/docker/Dockerfile\"."
+    )
+
+    // The following three settings control how the generated image is tagged. The image portion of
+    // image tags will be, for the main image:
+    //   ${imageRegistryHost}/${imageNamePrefix}/${imageName}
+    // and for the dependency image will be:
+    //   ${imageRegistryHost}/${imageNamePrefix}/${imageName}-dependency
+    //
+    // See the documentation for details on which tags will be used by `dockerBuild` and
+    // `dockerPush`.
+    val imageRegistryHost: SettingKey[String] = settingKey[String](
+      "The base name of the image you're creating. Defaults to " +
+        "allenai-docker-private-docker.bintray.io ."
```

Comment:
Constant used.

These are online help strings used by `inspect` in the sbt console (and possibly with other sbt commands?), so they should contain the defaults rather than point to code.

Documenting defaults at all is a little awkward, since the keys *technically* don't have a default until you enable the plugin. I think it's a good balance to have them documented here.

Anyone reading the plugin code will hopefully know enough to look at the `projectSettings` method.